ql/pricingengines/capfloor/blackcapfloorengine.cpp

```diff
@@ -145,7 +158,11 @@ namespace QuantLib {
 
         results_.additionalResults["optionletsPrice"] = values;
         results_.additionalResults["optionletsVega"] = vegas;
+        results_.additionalResults["optionletsDelta"] = deltas;
+        results_.additionalResults["optionletsDeflators"] = deflators;
```

Comment:
I'm not sure about the intermediate results.  They may be useful for debugging every once in a while, but you're copying a lot of vectors around.

Reply:
That is true, however I was thinking about exposing those vectors in Python. Those vectors are useful for some of the capital requirements calculations or other analytics in risk if you want to explain changes in P/L. The whole purpose of this PR was to expose `deltas`, but at the same time I am wandering if there would be some elegant way to expose to Python the whole structure of `arguments_` with some nice way of supporting in the IDE selection of available results. Regarding C++ I was also thinking if it wouldn't be usefull to add a method to `CapFloor` class, that would overload:

```c++
template <class T>
    inline T Instrument::result(const std::string& tag) const {
```
where instead of `string` argument one could pass a field of a class defined in `CapFloor::results` with relevant strings defined in the fields of that class. 
What do you think?

Reply:
I'm not sure I understand what you have in mind about `Instrument::result`.

About explaining changes in P&L: I see that, but in that case you want the basic pieces (accruals, notionals, discounts etc.) instead of the combined deflator, don't you?  And most of those pieces can be extracted from the leg independently of the engine...

Reply:
I have reduced the number of vectors copied to results. I hope that I used properly legs in the tests:
```
ext::shared_ptr<FloatingRateCoupon> c = ext::dynamic_pointer_cast<FloatingRateCoupon>(capLeg[n]);
accrualFactor = c->nominal() * c->accrualPeriod() * c->gearing();
```
Regarding my previous comment, I thought about adding a class within `CapFloor`, that would serve as a dictionary of available additionalResults, so that one wouldn't have to copy the strings with names of available results.

Reply:
Oh, I see.  You mean something like this?

    cap->result<CapFloor::Something>("more").discountFactors;

Reply:
Something like that, but now I see that of little value in C++ and for other languages proper SWIG code will do the job, so I guess there is no point in that.